internal/resources/cluster/tkgservicevsphere/resource_tkg_service_vsphere.go

```diff
@@ -337,16 +337,18 @@ func expandTKGSTopology(data []interface{}) (topology *tkgservicevspheremodel.Vm
 		return topology
 	}
 
-	topology = &tkgservicevspheremodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgservicevsphereTopology{}
 	topologyData, _ := data[0].(map[string]interface{})
+	topology = &tkgservicevspheremodel.VmwareTanzuManageV1alpha1ClusterInfrastructureTkgservicevsphereTopology{}
 
 	if v, ok := topologyData[controlPlaneKey]; ok {
-		topology.ControlPlane = expandTKGSTopologyControlPlane(v.([]interface{}))
+		if v1, ok := v.([]interface{}); ok {
```

Comment:
restrain from using v1 and short variables, it makes it confusing to read them.
can you change this variable which suits the context

Reply:
For the ease of understanding, have used cp(control plane) instead of v1 here. This variable naming is the issue in many other places where we are calling other expand functions. We can take this up later as a maintenance task and cover all such places. Here, I just followed the conventions we were using at other such points in the code.